[WIP] Add verification of Xcode project as part of precheck

KrauseFx opened the pull request:
To give this a try, do the following

- Use a sample Xcode project
- Add a build phase with the content, replace the path to your `bundle` binary (TODO, this should not be necessary)
```
FASTLANE_DISABLE_COLORS=1 /Users/fkrause/.rbenv/shims/bundle exec fastlane verify_firebase
```
- Update your `Gemfile` to point to  your local fastlane code base, with this branch checked out and run `bundle update`
- Currently we only check if we can find the plist file, and if it has some content (at least one row) and that it can be parsed, check out the [full class here](https://github.com/davidair/fastlane/blob/master/precheck/lib/precheck/rules/verify_plist_rule.rb#L40)
- To run it manually from outside of Xcode (useful for debugging reasons) you can use the following, with of course your actual Plist file path and Xcode project path. This way you can iterate faster, as it doesn't require you to re-compile the every every single time
```
PROJECT_FILE_PATH=./Themoji.xcodeproj INFOPLIST_FILE =./Themoji/Info.plist  bundle exec fastlane verify_firebase
```

KrauseFx committed:
Initial XcodeEnv class

KrauseFx committed:
Add run_as_build_phase? method

taquitos committed:
Version bump to 2.64.1 (#10893)

KrauseFx committed:
Add initial working implementation

KrauseFx committed:
Improve precheck docs (#10897)

taquitos committed:
When building plugin scores, we should ignore repos that are private (#10898)

- set GIT_TERMINAL_PROMPT=0

tattn committed:
remove an url encoding for a fragment identifier (#10886)

KrauseFx committed:
Add missing 1 in docs (#10890)

revolter committed:
Remove superfluous characters from code signing URL (#10900)

taquitos commented:
so clean

taquitos committed:
Failed logins can cause exception loop (#10899)

* Failed logins can cause exception loop

* Fix test break

* omg rubocop I love you and hate you

* make things a little more ruby-esque

davidair commented:
The environment variable to specify the plist is actually INFOPLIST_FILE. Can you please update the description?

taquitos commented:
@davidair our naming conventions generally favor `INFOPLIST_PATH` when talking about a path to a file

KrauseFx commented:
Updated 👍

davidair committed:
Introducing a project validator rule

KrauseFx commented on Gemfile:
```diff
@@ -2,6 +2,8 @@ source "https://rubygems.org"
 
 gem "xcode-install", ">= 2.2.1" # needed for running xcode-install related tests
 
+gem "xcodeproj", ">= 1.5.3"
```
This should already be a dependency of fastlane I think, check out `fastlane.gemspec` 👍

davidair committed:
Adding bundle id check, removing unneeded gem reference from Gemfile

scompt committed:
Allow PATCH method in github_api requests (#10908)

davidair committed:
Adding a passing test

davidair committed:
Adding Firebase Auth rule, restructuring code. Example usage: FASTLANE_DISABLE_COLORS=1 PROJECT_FILE_PATH=_path_ TARGET_NAME=_name_ CONFIGURATION=Release  bundle exec fastlane verify_firebase

revolter committed:
Fix broken link in code comments (#10911)

revolter committed:
Add warning comment to auto-generated docs files (#10910)

Commit:
Fixing a spelling error in the config file test where no value is provided (#10920)

KrauseFx committed:
Merge branch 'master' of https://github.com/davidair/fastlane into davidair-master

davidair committed:
Cleaning up directory structure

bstpierr committed:
Fix access to Xcode project info

Error messages were attempting to access undefined variables.

bstpierr committed:
Add verify-firebase.sh script

Also added the Build Script step to all projects to perform the checks
when built.

davidair committed:
Adding Dynamic Links validation

davidair committed:
Merge branch 'master' of github.com:davidair/fastlane

bstpierr committed:
Add output reducing environment variables

Remove all the non-precheck output I'm seeing

bstpierr committed:
Merge branch 'master' of https://github.com/davidair/fastlane

* 'master' of https://github.com/davidair/fastlane:
  Adding Dynamic Links validation

KrauseFx commented on precheck/spec/fixtures/Common/verify-firebase.sh:
```diff
@@ -0,0 +1,23 @@
+
+# Actually run fastlane from the fastlane root directory.
+cd ../../../
+bundle exec fastlane verify_firebase
```
In our experience, lots of people just use system Ruby and don't have rvm, and also some people use rbenv over rvm, do we need to add code to cover those also?

bstpierr committed:
Format Precheck output for Xcode

Also removed a debug statement that was using puts.

bstpierr committed:
Omit regular Precheck output from runner.rb when run from Xcode

This extra output does not conform to the warning: / error:  that Xcode
expects

bstpierr committed:
Force warning output for Xcode

The errors have an extra, awkward message about displaying an error
without a non-zero output

bstpierr committed:
Use Xcode messaging

Failure data has useful info for Xcode, rather than standard messaging.

KrauseFx committed:
Merge branch 'master' of https://github.com/davidair/fastlane into davidair-master

KrauseFx committed:
Remove CocoaPods dependency

KrauseFx committed:
Fix code style across all new files

bstpierr committed:
Fix undefined access to get_configuration variable

I ran bundle exec rubocop -a and didn't get issues after reverting this
back.

bstpierr committed:
Add path information to RuleReturn

This allows the Xcode warning to open a given file directly from the
warning.

davidair committed:
Fixing incorrect failures in rules

davidair committed:
Fixing lint issues

davidair committed:
Fixing a few bugs

davidair committed:
Adding a missing check, adding suggestion on improving applink check

kapfab commented:
`gym` seems to expect a project file in its prechecks even when it is requested to export from an existing archive.
As a workaround, we had to add a dumb Xcode project file to our lanes in order to able to export IPAs.
We will report an issue so that `skip_build_archive` option does not look for a useless project file, but I understand this PR will make things even boring for our case until this issue is fixed…

joshdholtz commented:
@KrauseFx You still working on this? 😊

KrauseFx commented:
Ok I don't think we'll go ahead with this, let's close it
